components/sftp/sftp.app.js

```diff
@@ -0,0 +1,70 @@
+module.exports = {
+  type: "app",
+  app: "sftp",
+  props: {
+    db: "$.service.db"
+  },
+  methods: {
+    config: {
+      host: null,
+      port: null,
+      username: null,
+      password: null,
+    },
+    async getClient(deps) {
+      const ssh2SftpClient = require('ssh2-sftp-client'); // src: https://github.com/theophilusx/ssh2-sftp-client/blob/master/src/index.js
+      const sftp = new ssh2SftpClient();
+
+      // dependency injection for integration tests
+      if (deps) {
+        this.config.host = deps.host;
+        this.config.port = deps.port;
+        this.config.username = deps.username;
+        this.config.password = deps.password;
+      }
+
+      // production - how auth is set
+      if (this.$auth && this.$auth.host) {
+        this.config.host = this.$auth.host;
+        this.config.port = this.$auth.port;
+        this.config.username = this.$auth.username;
+        this.config.password = this.$auth.password;
+      }
+
+      await sftp.connect(this.config);
+
+      return sftp;
+    },
+    async cwd() {
```

Comment:
Is there a specific reason why `cwd` doesn't accept an `sftp` instance, where `list` and other methods do?